src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherDocumentPreRules.java

```diff
@@ -67,11 +70,26 @@ public boolean doPrompts(Document document) {
 
         // Handle custom pre-rules:
         preRulesOK &= this.checkCheckStubTextOverflow((DisbursementVoucherDocument) document);
+        
+        setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(document);
 
         return preRulesOK;
     }
 
-    /**
+	private void setIncomeClassNonReportableForForeignVendorWithNoTaxReviewRequired(Document document) {
+		DisbursementVoucherDocument dvDoc = (DisbursementVoucherDocument) document;
+		DisbursementVoucherPayeeDetail dvPayeeDetail = dvDoc.getDvPayeeDetail();
+		
+		String payeeTypeCode = dvPayeeDetail.getDisbursementVoucherPayeeTypeCode();
+		String paymentReasonCode = dvPayeeDetail.getDisbVchrPaymentReasonCode();
+		Integer vendorHeaderId = dvPayeeDetail.getDisbVchrVendorHeaderIdNumberAsInteger();
+
+		if (SpringContext.getBean(CuDisbursementVoucherTaxServiceImpl.class).isForeignVendoAndNOTaxReviewRequired(payeeTypeCode,paymentReasonCode, vendorHeaderId)) {
```

Comment:
Another place where SpringContext.getBean() could be extracted into a getter for better testability.